gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -240,22 +215,47 @@ int main(int argc, char* argv[]) {
       return 3;
     }
 
-    auto proto = bhive_importer.BasicBlockProtoFromMachineCode(*bytes);
+    llvm::Expected<std::vector<gematria::DisassembledInstruction>>
+        instructions = gematria::DisassembleAllInstructions(
+            llvm_support->mc_disassembler(), llvm_support->mc_instr_info(),
+            llvm_support->mc_register_info(), llvm_support->mc_subtarget_info(),
+            *inst_printer, 0, *bytes);
 
     // Check for errors.
-    if (!proto.ok()) {
-      std::cerr << "Failed to disassemble block '" << hex << ": "
-                << proto.status() << "\n";
+    if (!instructions) {
+      std::cerr << "Failed to disassemble block '" << hex << "\n";
       continue;
     }
 
+    // Get used registers
+    std::map<unsigned, bool> used_registers;
+    for (const gematria::DisassembledInstruction& instruction : *instructions) {
+      for (unsigned operand_index = 0;
+           operand_index < instruction.mc_inst.getNumOperands();
+           ++operand_index) {
+        if (instruction.mc_inst.getOperand(operand_index).isReg()) {
+          unsigned register_number =
+              instruction.mc_inst.getOperand(operand_index).getReg();
+          if (register_number == 0) continue;
+          unsigned super_register_number = 0;
+          for (MCPhysReg testing :
+               reg_info.superregs_inclusive(register_number)) {
+            super_register_number = testing;
+          }
+          used_registers[super_register_number] = true;
+        }
+      }
+    }
```

Comment:
Thinking about it a bit more, this should go to a library function where we can also add proper unit tests.

Reply:
Good point. Split into #71, where the actual functionality is implemented.